### PR TITLE
Add pigweed support for nrfconnect

### PR DESCRIPTION
**Pull request by jlatusek**
Related to #29682

## Changes
Add a call of the wrapper for running all Pigweed tests in test_driver for nrfconnect and link the necessary library.

## Testing

CI will test it. In related PR, there are changes for all platforms. The CI there passes all checks.

### CODE REVIEW

File: src/test_driver/nrfconnect/main/runner.cpp [hunk 1]
```diff
@@ -29,13 +30,24 @@ using namespace ::chip::DeviceLayer;
 
 LOG_MODULE_REGISTER(runner, CONFIG_MATTER_LOG_LEVEL);
 
-extern "C" int main(void)
+extern "C" {
+void console_init(void) {}
```
Comment (Damian-Nordic) [inline src/test_driver/nrfconnect/main/runner.cpp:34]: This is a hack that shadows Zephyr function definitions. Could you at least add a TODO comment that the tests should rather configure and use Zephyr's console subsystem?

File: src/test_driver/nrfconnect/main/runner.cpp [hunk 2]
```diff
@@ -29,13 +30,24 @@ using namespace ::chip::DeviceLayer;
...
+    return 0;
+}
+
+int main(void)
 {
     VerifyOrDie(settings_subsys_init() == 0);
 
     LOG_INF("Starting CHIP tests!");
     int status = RunRegisteredUnitTests();
+    status += chip::test::RunAllTests();
```
Comment (Damian-Nordic) [inline src/test_driver/nrfconnect/main/runner.cpp:48]: Is the plan to completely replace the old method of registering UTs or we're going to support both in parallel?

Comment (jlatusek) [reply src/test_driver/nrfconnect/main/runner.cpp:48]: The plan is to replace the old method completely successively test after test. The old method will be removed at the end of the process.

Comment (Damian-Nordic) [reply src/test_driver/nrfconnect/main/runner.cpp:48]: Thanks for the info!

File: src/test_driver/nrfconnect/main/runner.cpp (hunk 1 above)
Comment (jlatusek) [reply src/test_driver/nrfconnect/main/runner.cpp]: I've checked, and this hack was necessary in the first version of my changes. This is no longer required, so I removed it.

Comment (Damian-Nordic) [reply src/test_driver/nrfconnect/main/runner.cpp]: ok, thanks! :)

**Review by Damian-Nordic — APPROVED** [review]

**Review by arkq — APPROVED** [review]

**Pull request merged**